packages/flame/lib/src/components/input/hud_button_component.dart

```diff
@@ -6,8 +6,15 @@ import '../../../extensions.dart';
 import '../../../input.dart';
 import '../../gestures/events.dart';
 
+/// The [HudButtonComponent] bundles two [PositionComponent]s, one that shows
+/// when the button is being pressed, and one that shows otherwise.
+///
+/// Note: if you are setting the [button] in [onLoad] instead of passing it in
+/// through the constructor, be sure that you add it to the component too by
+/// simply doing `add(button);` and that you also set the [button] variable to
+/// your [PositionComponent] with `button = yourPositionComponent;`.
```

Comment:
Would be nice if we could prevent that instead, like we do on the the flame game.

Reply:
Where/how do we do that on `FlameGame`?

Reply:
Iirc, on the prepareComponent method, should be only assertion left there

Reply:
Hmm, not sure I follow how we would do it with `prepareComponent`?
I could have a check and assert in `onMount` and add it if it is not added though?